### PR TITLE
Implemet_decorate_pattern

### PR DESCRIPTION
**Pull request by aliveGUY**
# Implemented decorate pattern :checkered_flag: 
-Created `Nameable` component class
- Created base `Decorator` class
- - Inherits from `Nameable`
-  Created `CapitalizeDecorator` class
- - Inherits from `Decorator`
- Created `TrimmerDecorator` class
- - Inherits from `Decorator`

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    name
+  end
+end
```
Comment (andersonlebon) [inline nameable.rb:5]: - [ ] Kindly make sure your  correct_name method can  raise an error: for example

```rb
 def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
```

File: person.rb [hunk 2]
```diff
@@ -1,5 +1,8 @@
-class Person
+require './nameable'
+
+class Person < Nameable
   def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
```
Comment (andersonlebon) [inline person.rb:8]: - [ ] Kindly your Person class should contain a method `correct_name`

File: trimmer_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0..10]
```
Comment (andersonlebon) [inline trimmer_decorator.rb:5]: - [ ] Kindly the name has a maximum of 10 characters. this syntax will correct the name with a maximum of 11 characters, Kindly use ` @nameable.correct_name[0...10]` or ` @nameable.correct_name[0..9`

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @aliveGUY 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Rustamxon7 — APPROVED** [review]
## Hello @aliveGUY👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**